docs/manual/generics.tex

```diff
@@ -156,58 +156,104 @@
 If the \<extends> clause is omitted,
 then the upper bound defaults to \<@\emph{TopType} Object>.
 \item
-If the \<extends> clause is written but contains no type qualifier,
+If the \<extends> clause is \<extends Object>
+(without an explicit type qualifier in the given type system),
+then the upper bound also defaults to \<@\emph{TopType} Object>.
+\item
+If the \<extends> clause is written and contains no type qualifier and the
+upper bound is not \<Object>,
 then the normal defaulting rules apply to the type in the \<extends>
 clause (see Section~\ref{climb-to-top}).
 \end{itemize}
 
-The upper-bound rules mean that even though in Java the following two
-declarations are equivalent:
+In Java the following two declarations are equivalent
+(they permit every instantiation):
 
 \begin{Verbatim}
   class MyClass<T>
   class MyClass<T extends Object>
 \end{Verbatim}
 
 \noindent
-they specify different type qualifiers on the upper bound,
-if the type system's default annotation is not its top annotation.
+and the upper-bound rules mean that a pluggable type system also
+treats them identically, permitting every instantiation since the upper
+bound is the top type qualifier.
 
 The Nullness type system is an example.
 
 \begin{Verbatim}
   class MyClass<T>                 ==  class MyClass<T extends @Nullable Object>
-  class MyClass<T extends Object>  ==  class MyClass<T extends @NonNull Object>
+  class MyClass<T extends Object>  ==  class MyClass<T extends @Nullable Object>
+  class MyClass<T extends Date>    ==  class MyClass<T extends @NonNull Date>
 \end{Verbatim}
 
-The rationale for this choice is:
-\begin{itemize}
-\item
-  The ``\code{<T>}'' in \code{MyClass<T>} means ``fully unconstrained'',
-  and the rules maintain that, without the need for a programmer to
-  change existing code.
-\item
-  The ``\code{Object}'' in \code{MyClass<T extends Object>} is treated
-  exactly like every other occurrence of \code{Object} in the program ---
-  it would be confusing for different occurrences of \code{Object} to mean
-  different annotated types.
-\end{itemize}
+An unfortunate consequence of the special case for \code{<T extends
```

Comment:
I think for me, there are so many exceptions to `Object` == `@NonNull Object`, I've never thought this was confusing.  (Casts are another place not mentioned here.) The more unfortunate consequence is that `T extends Object` and `T extends Date` are treated differently.